x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/ProfilingIndexManager.java

```diff
@@ -190,4 +182,54 @@ public void onFailure(Exception e) {
             );
         });
     }
+
+    static class ManagedIndex {
```

Comment:
So, this name confused me a bit, I thought a managed index meant managed by ILM. But If I understand correctly, only kv indices are managed by ILM right?  Can we add some comments here that explain what this is and how it's used?

Reply:
Naming is hard. :) Its purpose is to provide a type for indices that are created by `ProfilingIndexManager`. Maybe `ProfilingIndex` is a better name. I'll update the name and add a comment in the code.

> only kv indices are managed by ILM right?

Yes, the regular ones are only created but are not managed by ILM.